packages/dd-trace/test/appsec/iast/analyzers/template-injection-analyzer.handlebars.plugin.spec.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const { prepareTestServerForIast } = require('../utils')
+const { storage } = require('../../../../../datadog-core')
+const iastContextFunctions = require('../../../../src/appsec/iast/iast-context')
+const { newTaintedString } = require('../../../../src/appsec/iast/taint-tracking/operations')
+
+describe('template-injection-analyzer with handlebars', () => {
+  withVersions('handlebars', 'handlebars', version => {
+    let lib, badSource, goodSource
+    before(() => {
+      lib = require(`../../../../../../versions/handlebars@${version}`).get()
+      badSource = `
+          {{#with "s" as |string|}}
+            {{#with "e"}}
+              {{#with split as |conslist|}}
+                {{this.pop}}
+                {{this.push (lookup string.sub "constructor")}}
+                {{this.pop}}
+                {{#with string.split as |codelist|}}
+                  {{this.pop}}
+                  {{this.push "return JSON.stringify(process.env);"}}
+                  {{this.pop}}
+                  {{#each conslist}}
+                    {{#with (string.sub.apply 0 codelist)}}
+                      {{this}}
+                    {{/with}}
+                  {{/each}}
+                {{/with}}
+              {{/with}}
+            {{/with}}
+          {{/with}}
+          `
+      goodSource = '<p>{{name}}</p>'
```

Comment:
For vulnerability detection, the difference about the string is not its content, it is the source. You can use this `<p>{{name}</p>` for vulnerable and not vulnerable tests. It should be tainted in the vulnerable tests and not tainted in not vulnerable tests.